hail/src/main/scala/is/hail/expr/ir/agg/CollectAsSetAggregator.scala

```diff
@@ -144,7 +144,8 @@ class AppendOnlySetState(val fb: EmitFunctionBuilder[_], t: PType) extends Point
 class CollectAsSetAggregator(t: PType) extends StagedAggregator {
   type State = AppendOnlySetState
 
-  val resultType: PSet = PSet(t)
+  assert(PType.canonical(t) == t)
```

Comment:
same question as on the other PR, why not check virtual type identity (we could keep using isOfType which now checks virtual type identity, if you wanted to have an easier find/replace when we add restriction compatibility checks)

Reply:
The canonical checks can be isCanonical

Reply:
`assert(isCanonical(t))`?